clientconn_test.go

```diff
@@ -742,6 +742,7 @@ func (s) TestClientUpdatesParamsAfterGoAway(t *testing.T) {
 		}
 		if ctx.Err() != nil {
 			// Timeout
+			cc.mu.RUnlock()
```

Comment:
How about moving the `RUnlock()` to line 738 and remove the other two calls to it on 740 and 748?

Reply:
It gets locked every 10ms on line 736, so I would assume you would want to unlock after each check? Moving to 730 still has the hang on failing tests

Reply:
> Moving to 730 still has the hang on failing tests

I suggested moving to 738 not 730.  Maybe you misread?

Reply:
Ah! Yes you're right, I misread. That seems like a better solution. Updated.

Reply:
No worries, thank you for the fix!